### ISSUE TITLE
[Android] App crashed totally after call `request()` function when `minifyEnabled` is `true`

### ISSUE THREAD

**Issue by coder-xiaomo** [zoontek/react-native-permissions#855]
### Before submitting a new issue

- [X] I tested using the latest release of the library, as maybe the bug has been already fixed.
- [X] I checked for possible duplicate issues, with possible answers.

### Bug summary

I am using React Native 0.73.5 (on Android device) with react-native-permissions 4.1.4

It works fine when the `android->buildTypes->release->minifyEnabled` is false `app/build.gradle` (default is `false`)

But **since I set minifyEnabled to `true`, app crashed totally after I call `request()` function.**

i guess the class `RNPermissions` was obfuscated.

I tried add these lines into `proguard-rules.pro` file, but it seems **not work**

```bash
-keepattributes StackMapTable
-dontoptimize
-dontobfuscate

-dontwarn com.zoontek.rnpermissions.**
-keep class com.zoontek.rnpermissions.** { *; }

-keep class com.zoontek.rnpermissions.RNPermissionsPackage
-keep class java.util.** { *; }
-keep class com.android.installreferrer.** { *; }
```

I don't know exactly what went wrong, is there any solution to solve this problem.

Thanks very much!


### Library version

4.1.4

### Environment info

```shell
info Fetching system and libraries information...
System:
  OS: Windows 11 10.0.22631
  CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
  Memory: 1.71 GB / 15.83 GB
Binaries:
  Node:
    version: 18.18.0
    path: D:\Program\Development\Environment\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: ~\AppData\Roaming\npm\yarn.CMD
  npm:
    version: 10.2.5
    path: ~\AppData\Roaming\npm\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK:
    API Levels:
      - "29"
      - "31"
      - "33"
      - "34"
    Build Tools:
      - 28.0.3
      - 30.0.3
      - 33.0.1
      - 34.0.0
    System Images:
      - android-33 | Google APIs Intel x86_64 Atom
      - android-34 | Google Play Intel x86_64 Atom
      - android-UpsideDownCake-ext5 | Google APIs Intel x86 Atom_64
    Android NDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
IDEs:
  Android Studio: Not Found
  Visual Studio:
    - 17.8.34408.163 (Visual Studio Enterprise 2022)
Languages:
  Java: 11.0.18.14
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.5
    wanted: 0.73.5
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```


### Steps to reproduce

1. Set `android->buildTypes->release->minifyEnabled` to true `app/build.gradle` (default is `false`)
2. call request() to request
3. App crashed with following log

here is the app crash logs:

```
2024-03-08 16:17:50.138 16453-16489 EGL_emulation           com.example.app                     D  app_time_stats: avg=192.72ms min=3.42ms max=3034.30ms count=17
2024-03-08 16:17:50.240 16453-16502 ReactNativeJS           com.example.app                     E  Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'RNPermissions' could not be found. Verify that a module by this name is registered in the native binary.Bridgeless mode: false. TurboModule interop: false. Modules loaded: {"NativeModules":["HeadlessJsTaskSupport","UIManager","PlatformConstants","DeviceInfo","DeviceEventManager","RNVectorIcons","RNLocalize","SoundManager","I18nManager","NativeAnimatedModule","ImageLoader","Appearance","BlobModule","Networking","Timing","RNCSafeAreaContext","AppState","StatusBarManager","StatusBarManager","IntentAndroid","AccessibilityInfo","FileReaderModule"],"TurboModules":[],"NotFound":["NativePerformanceCxx","NativePerformanceObserverCxx","RedBox","BugReporting","LinkingManager","FrameRateLogger","KeyboardObserver","ModalManager","AccessibilityManager","RNPermissions"]}, js engine: hermes
2024-03-08 16:17:50.248 16453-16503 AndroidRuntime          com.example.app                     E  FATAL EXCEPTION: mqt_native_modules
                                                                                                    Process: com.example.app, PID: 16453
                                                                                                    com.facebook.react.common.JavascriptException: Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'RNPermissions' could not be found. Verify that a module by this name is registered in the native binary.Bridgeless mode: false. TurboModule interop: false. Modules loaded: {"NativeModules":["HeadlessJsTaskSupport","UIManager","PlatformConstants","DeviceInfo","DeviceEventManager","RNVectorIcons","RNLocalize","SoundManager","I18nManager","NativeAnimatedModule","ImageLoader","Appearance","BlobModule","Networking","Timing","RNCSafeAreaContext","AppState","StatusBarManager","StatusBarManager","IntentAndroid","AccessibilityInfo","FileReaderModule"],"TurboModules":[],"NotFound":["NativePerformanceCxx","NativePerformanceObserverCxx","RedBox","BugReporting","LinkingManager","FrameRateLogger","KeyboardObserver","ModalManager","AccessibilityManager","RNPermissions"]}, js engine: hermes, stack:
                                                                                                    invariant@1:65375
                                                                                                    getEnforcing@1:65218
                                                                                                    anonymous@1:1248552
                                                                                                    loadModuleImplementation@1:45996
                                                                                                    guardedLoadModule@1:45538
                                                                                                    metroRequire@1:45160
                                                                                                    anonymous@1:1246972
                                                                                                    loadModuleImplementation@1:45996
                                                                                                    guardedLoadModule@1:45538
                                                                                                    metroRequire@1:45160
                                                                                                    anonymous@1:1245899
                                                                                                    loadModuleImplementation@1:45996
                                                                                                    guardedLoadModule@1:45495
                                                                                                    metroRequire@1:45160
                                                                                                    ?anon_0_@1:1244769
                                                                                                    asyncGeneratorStep@1:524748
                                                                                                    _next@1:525005
                                                                                                    anonymous@1:524957
                                                                                                    tryCallTwo@61:8
                                                                                                    doResolve@216:24
                                                                                                    Promise@82:13
                                                                                                    anonymous@1:524878
                                                                                                    requestNotificationPermission@1:1244988
                                                                                                    onPress@1:1160182
                                                                                                    anonymous@1:962362
                                                                                                    _performTransitionSideEffects@1:392527
                                                                                                    _receiveSignal@1:392117
                                                                                                    onResponderRelease@1:390921
                                                                                                    invokeGuardedCallbackImpl@1:315008
                                                                                                    invokeGuardedCallback@1:315065
                                                                                                    invokeGuardedCallbackAndCatchFirstError@1:315097
                                                                                                    executeDispatch@1:315226
                                                                                                    executeDispatchesAndReleaseTopLevel@1:319436
                                                                                                    forEachAccumulated@1:316717
                                                                                                    anonymous@1:319799
                                                                                                    batchedUpdatesImpl@1:379099
                                                                                                    batchedUpdates@1:319352
                                                                                                    _receiveRootNodeIDEvent@1:319637
                                                                                                    receiveTouches@1:373867
                                                                                                    __callFunction@1:71173
                                                                                                    anonymous@1:69478
                                                                                                    __guard@1:70434
                                                                                                    callFunctionReturnFlushedQueue@1:69436
                                                                                                    
                                                                                                    	at com.facebook.react.modules.core.ExceptionsManagerModule.reportException(ExceptionsManagerModule.java:65)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
                                                                                                    	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:149)
                                                                                                    	at com.facebook.jni.NativeRunnable.run(Native Method)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:234)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
2024-03-08 16:17:50.431 16453-16527 TrafficStats            com.example.app                     D  tagSocket(125) with statsTag=0xffffffff, statsUid=-1
2024-03-08 16:17:50.609 16453-16523 TrafficStats            com.example.app                     D  tagSocket(5) with statsTag=0xffffffff, statsUid=-1
2024-03-08 16:17:50.891 16453-16483 TrafficStats            com.example.app                     D  tagSocket(73) with statsTag=0x8001, statsUid=-1
2024-03-08 16:17:51.143 16453-16489 EGL_emulation           com.example.app                     D  app_time_stats: avg=8.99ms min=4.08ms max=28.79ms count=56
2024-03-08 16:17:52.159 16453-16489 EGL_emulation           com.example.app                     D  app_time_stats: avg=4.98ms min=3.66ms max=7.14ms count=61
2024-03-08 16:17:52.280   568-1750  ActivityTaskManager     system_server                        W    Force finishing activity com.example.app/.MainActivity
2024-03-08 16:17:52.293 16453-16503 Process                 com.example.app                     I  Sending signal. PID: 16453 SIG: 9
2024-03-08 16:17:52.351   568-1366  ActivityManager         system_server                        I  Process com.example.app (pid 16453) has died: fg  TOP 
2024-03-08 16:17:52.357   568-1363  WindowManager           system_server                        I  WIN DEATH: Window{5436622 u0 com.example.app/com.example.app.MainActivity}
2024-03-08 16:17:52.359   568-719   ConnectivityService     system_server                        D  releasing NetworkRequest [ REQUEST id=252, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VCN_MANAGED Uid: 10201 RequestorUid: 10201 RequestorPkg: com.example.app UnderlyingNetworks: Null] ] (release request)
2024-03-08 16:17:52.360   568-1363  InputManager-JNI        system_server                        W  Input channel object '5436622 com.example.app/com.example.app.MainActivity (client)' was disposed without first being removed with the input manager!
2024-03-08 16:17:52.365   568-1750  ImeTracker              system_server                        I  com.example.app:bb51c250: onRequestHide at ORIGIN_SERVER_HIDE_INPUT reason HIDE_REMOVE_CLIENT
2024-03-08 16:17:52.365   568-1750  ImeTracker              system_server                        I  com.example.app:bb51c250: onCancelled at PHASE_SERVER_SHOULD_HIDE
2024-03-08 16:17:52.459   568-596   WindowManager           system_server                        W  Failed to deliver inset control state change to w=Window{5436622 u0 com.example.app/com.example.app.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:584)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:479)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3802)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:343)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$8yykPRG1GyNq_J17QvL9d5xANMc(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda2.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:294)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:202)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$aHNu1uhcqxihX5NZc4McDDQPAyw(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1337)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1348)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:952)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:878)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1322)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:67)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
2024-03-08 16:17:52.647   568-592   WindowManager           system_server                        V      info={id=99 t=CLOSE f=0x10 trk=0 r=[0@Point(0, 0)] c=[{WCT{RemoteToken{29aef49 Task{48fe3c #1 type=home}}} m=TO_FRONT f=SHOW_WALLPAPER|MOVE_TO_TOP leash=Surface(name=Task=1)/@0x6da57c sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},{WCT{RemoteToken{741db4b Task{1c42536 #109 type=standard A=10201:com.example.app}}} m=CLOSE f=NONE leash=Surface(name=Task=109)/@0xf4b22c5 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},{null m=TO_FRONT f=IS_WALLPAPER leash=Surface(name=WallpaperWindowToken{d827de4 token=android.os.Binder@a43638})/@0xc72af53 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0}]}
2024-03-08 16:17:52.764   568-596   WindowManager           system_server                        W  Failed to deliver inset control state change to w=Window{5436622 u0 com.example.app/com.example.app.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:584)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:479)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3802)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:343)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$8yykPRG1GyNq_J17QvL9d5xANMc(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda2.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:294)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:202)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$aHNu1uhcqxihX5NZc4McDDQPAyw(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1337)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1348)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:952)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:878)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1322)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:67)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
---------------------------- PROCESS ENDED (16453) for package com.example.app ----------------------------
```

### Reproducible sample code

```js
android/app/build.gradle


apply plugin: "com.android.application"
apply plugin: "org.jetbrains.kotlin.android"
apply plugin: "com.facebook.react"

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = true // default: false
def jscFlavor = 'org.webkit:android-jsc:+'

android {
    ndkVersion rootProject.ext.ndkVersion
    // buildToolsVersion rootProject.ext.buildToolsVersion
    compileSdk rootProject.ext.compileSdkVersion
    namespace "com.example.app"
    defaultConfig {
        applicationId "com.example.app"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
    }
    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }
    buildTypes {
        debug {

            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            crunchPngs false
        }
    }
}

dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    implementation("com.facebook.react:flipper-integration")

    if (hermesEnabled.toBoolean()) {
        implementation("com.facebook.react:hermes-android")
    } else {
        implementation jscFlavor
    }
}
```

android/app/proguard-rules.pro

```
# Add project specific ProGuard rules here.
# By default, the flags in this file are appended to flags specified
# in /usr/local/Cellar/android-sdk/24.3.3/tools/proguard/proguard-android.txt
# You can edit the include path and order by changing the proguardFiles
# directive in build.gradle.
#
# For more details, see
#   http://developer.android.com/guide/developing/tools/proguard.html

# Add any project specific keep options here:

-keepattributes StackMapTable
-dontoptimize
-dontobfuscate

-dontwarn com.zoontek.rnpermissions.**
-keep class com.zoontek.rnpermissions.** { *; }

-keep class com.zoontek.rnpermissions.RNPermissionsPackage
-keep class java.util.** { *; }
-keep class com.android.installreferrer.** { *; }
```
```

**Comment by imanshul** [comment]
I'm also facing same issue when i did a codepush to the user in release build, as soon as it reflects same crash occured

> 'RNPermissionsModule' could not be found. Verify that a module by this name is registered in the native binary.Bridgeless mode: false. TurboModule interop: false. Modules loaded: {"NativeModules":["HeadlessJsTaskSupport","UIManager","PlatformConstants","DeviceInfo","DeviceEventManager","Timing","RNVectorIcons","NativeAnimatedModule","SoundManager","ImageLoader","I18nManager","RNDatePicker","RNCWebView","ReactNativeBlobUtil","RNShare","RNCClipboard","RNCSafeAreaContext","AppState"],"TurboModules":[],"NotFound":["NativePerformanceCxx","NativePerformanceObserverCxx","RedBox","BugReporting","RNPermissionsModule"]}, js engine: hermes

**Comment by zoontek** [comment]
Why is your TurboModule interop to false?

**Comment by zhang-brook** [comment]
@zoontek I'm sorry, **there's nothing wrong with the project**, it's my mistake.

I create a new bare project, and install react-native-permissions, with the minifyEnabled set to true, and anything goes well.

so i recheck all my dependencies, its the problem of `react-native-update` (<10.3.0), after communicating with the author, the problem solved in react-native-update@10.3.0

@imanshul I don't know if you have the same problem, but I suspect it may also be due to the dependence on hot updates.

Sorry to bother 🙏
